agent/proxycfg/state.go

```diff
@@ -265,6 +276,9 @@ func (s *state) Watch() (<-chan ConfigSnapshot, error) {
 
 // Close discards the state and stops any long-running watches.
 func (s *state) Close(failed bool) error {
+	if s.stoppedRunning() {
```

Comment:
Should a double-close happen, this prevents it from doing double work at least (or trying to).